c_plus_plus/src/model/media_player/subtitles/srt_file.cpp

```diff
@@ -0,0 +1,47 @@
+#include "srt_file.h"
+
+#include "model/utils/time.h"
+
+#include <QFile>
+#include <QRegularExpression>
+#include <QTextStream>
+
+namespace Model
+{
+std::vector<SubtitleItem> SrtFile::parse(const QString &path)
+{
+    std::vector<SubtitleItem> subtitles;
+
+    QFile file(path);
+    if (file.open(QIODevice::ReadOnly | QIODevice::Text))
+    {
+        QTextStream in(&file);
+        in.setCodec("UTF-8");
+
+        QString content = in.readAll();
+
+        QRegularExpression patternStr(
+            R"((\d+).*?\n(\d{2}):(\d{2}):(\d{2}),(\d{3}) --> (\d{2}):(\d{2}):(\d{2}),(\d{3}).*?\n([\s\S]*?)\n\n)");
+
+        QRegularExpressionMatchIterator it = patternStr.globalMatch(content);
```

Comment:
const ref?

Reply:
Since we call `it.next()`, it can't be const.